src/XRP/core-xrpl-client.ts

```diff
@@ -334,10 +341,21 @@ export default class CoreXrplClient implements CoreXrplClientInterface {
    */
   private getFinalTransactionStatus(
     rawTransactionStatus: RawTransactionStatus,
+    lastLedgerPassed: boolean,
   ): TransactionStatus {
+    // I don't know if this is actually possible... just making the compiler happy in safest way
```

Comment:
I don't know if we can (or at least, if we can in a worthwhile way) because we still need the granularity of investigating whether it starts with a `tes` or not later on in this function....  I decided to take care of this by altering `RawTransactionStatus` to throw an error if the transaction response doesn't contain a result code (therefore that field can't later be undefined), because that indeed should be an error case and we should catch it at the source.